boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/iam/util/IAMRestClient.java

```diff
@@ -0,0 +1,266 @@
+
+            List<NameValuePair> params = new ArrayList<>();
+            params.add(new BasicNameValuePair(USER_KEY, username));
+            params.add(new BasicNameValuePair(SECRET_KEY, String.valueOf(password)));
+
+            if (asJWT) {
+                params.add(new BasicNameValuePair(TOKEN_KEY, TOKEN_VALUE));
+            }
+
+            httpPost.setEntity(new UrlEncodedFormEntity(params));
+
+            HttpResponse httpResponse = httpClient.execute(httpPost);
+
+            params = new ArrayList<>();
+
+            if ((httpResponse.getStatusLine().getStatusCode() != HttpStatus.SC_CREATED) ||
+                (null == httpResponse.getEntity())) {
+
+                throw new IAMException("Failed to acquire SSO ticket for the user '" + username,
+                                       new HTTPException(httpResponse.getStatusLine().getStatusCode()));
+            }
+
+            LOG.debug("IAM response: " + httpResponse);
+
+            String iamResponse = EntityUtils.toString(httpResponse.getEntity());
+
+            if (!asJWT) {
+                Document doc = Jsoup.parse(iamResponse);
+                String htmlSsoTicket = doc.getElementsByAttribute(ticketMarker).first().attributes().get(ticketMarker);
+
+                if (htmlSsoTicket != null) {
+
+                    Matcher matcher = pattern.matcher(htmlSsoTicket);
+
+                    if (matcher.find()) {
+                        ssoTicket = matcher.group(1);
+                    }
+                }
+            } else {
+                ssoTicket = iamResponse;
+            }
+
+            CommonUtils.assertNotNull(ssoTicket, "Unable to extract SSO ticket form IAM response. SSO ticket is null.");
+
+            LOG.debug("SSO ticket successfully generated for user " + username + "\n Token: " + ssoTicket);
+
+        } catch (Exception e) {
+            throw new IAMException("Error occurred while acquiring SSO ticket for the user " + username, e);
+        }
+
+        return ssoTicket;
+    }
+
+    /**
+     * acquires a service token from IAM microservice, based on a SSO ticket.
+     *
+     * @param url URL of IAM server to which the request is addressed
+     * @param service service concerned by the ticket
+     * @return service ticket
+     * @since version 8.3.0
+     */
+    public String getServiceToken(String url, String service) {
+
+        String token;
+
+        try (CloseableHttpClient httpClient = new CommonHttpClientBuilder().allowAnyCertificate(true).build()) {
+
+            HttpPost httpPost = new HttpPost(url);
+
+            List<NameValuePair> params = new ArrayList<>();
+            params.add(new BasicNameValuePair(SERVICE_KEY, service));
+            httpPost.setEntity(new UrlEncodedFormEntity(params));
+
+            HttpResponse httpResponse = httpClient.execute(httpPost);
+
+            if ((httpResponse.getStatusLine().getStatusCode() != HttpStatus.SC_OK) ||
+                (null == httpResponse.getEntity()))
```

Comment:
Having an if with no brackets is not within our code practice generally, don't you think?

Reply:
I agree. Fixed.